### PR TITLE
Numbafy split by containment

### PR DESCRIPTION
**Pull request by WenzDaniel**
**What is the problem / what does the code in this PR do**
In this PR we numbafy `split_by_containment` and change its return value into a numba.typed.List instead of a normal Python list. This has the advantage that the result of `split_by_containment` can be used in numba njitted functions which enhances there speed. Further, this PR increases the performance of  `split_by_containment` by a factor of x2 to x4 depending on its inputs. This performance increase will directly manifest in a performance increase of LoopPlugins. 
If the numba.typed.List output is used instead of a LoopPlugin one can gain another factor x10 boost if needed. This enables maybe new usages of `split_by_containment`. 
In addition to the function changes I added small tests to ensure, that the required replacement functions behave as their numpy counterparts. 

**Can you give a minimal working example (or illustrate with a figure)?**
A simple comparison notebook can be found [here](https://github.com/XENONnT/analysiscode/blob/master/StraxTests/NumbafiedSplitByContainment.ipynb).

### CODE REVIEW

**Review by jorana — APPROVED** [review]
Looks like absolutely great work Daniel.

Anything you want me to review specifically, I looked over it and it's clear you did your work with diligence.

I think I need to start paying more attention to the way you write the tests, it looks great, can you perhaps help me with even adding some comments there.

File: strax/processing/general.py [hunk 1]
```diff
@@ -135,13 +137,20 @@ def _fc_in(a_starts, b_starts, a_ends, b_ends, result):
...
-    """Return list of thing-arrays contained in each container
+    """
+    Return list of thing-arrays contained in each container. Result is
+    returned as a numba.typed.List or list if containers are empty.
 
-    Assumes everything is sorted, and containers are nonoverlapping
+    Assumes everything is sorted, and containers are non-overlapping.
     """
     if not len(containers):
         return []
```
Comment (jorana) [inline strax/processing/general.py:147]: Should this be a typed empy list?

Comment (WenzDaniel) [reply strax/processing/general.py]: Yes, thanks for spotting. It took a small hack to define the list type.

**Comment by WenzDaniel** [comment]
Thanks a lot. I think I addressed all your comments.

**Pull request merged**